### PR TITLE
Revamp push.md to include instructions on how to actually get set up

### PR DESCRIPTION
**Pull request by AlureonTheVirus**
Hopefully this is helpful to a lot of people. I have not documented the response types all that much buy I've included a TON of info on how to get set up so that people can at least start messing around with WebSockets and the API.

### CODE REVIEW

File: push.md [hunk 1]
```diff
@@ -1,27 +1,201 @@
...
+it may be hard to figure out what's required and what's not.
 
 Here is a list of some known WebSocket types and what they correspond to:
 
 * `ping` -> A ping from the web socket server
 * `line.create` -> New Group Message. This includes anything that makes a message such as calendar events,
   joining/leaving events, polls, etc.
 * `like.create` -> Someone likes a message that is yours in a group
-* `favorite` -> Someone likes a message that is not yours in a group
+* `favorite` -> Someone likes a message that is not yours in a group (can not be caught in the `/user/:user_id`, only in group or direct_message channels)
```
Comment (not-so-smart) [inline push.md]:
```suggestion
* `favorite` -> Someone likes a message that is not yours in a group (can not be caught in the `/user/:user_id` channel, only in group or direct_message channels)
```

File: push.md [hunk 2]
```diff
@@ -1,27 +1,201 @@
...
+You may want to read its docs for a deeper understanding of how it works.
+The [official documentation](https://dev.groupme.com/tutorials/push) has some outdated and incomplete information,
+it may be hard to figure out what's required and what's not.
 
 Here is a list of some known WebSocket types and what they correspond to:
 
 * `ping` -> A ping from the web socket server
 * `line.create` -> New Group Message. This includes anything that makes a message such as calendar events,
   joining/leaving events, polls, etc.
 * `like.create` -> Someone likes a message that is yours in a group
```
Comment (not-so-smart) [inline push.md:14]: How do `like.create` and `favorite` work in the era of reactions? Are these still called `like` and `favorite` despite containing reaction data?

File: push.md [hunk 3]
```diff
@@ -1,27 +1,201 @@
...
+(e.g., new messages in groups you're a part of or reactions attached to messages you have sent).
+
+The other two less important channel types are `/groups/:group_id`, and `/direct_message/:dm_id`. 
+Both send information about specific channels that a client wouldn't need to send push notifications for but would need to render the channel when it's open correctly and on the screen. 
+(e.g., typing indicators, new reactions attached to other messages in the channel that you have not sent, or even membership and role updates for specific users)
+
+## Subscribing to the `/user/:user_id` channel
+
+By far the simplest option for establishing a websocket connection is to use a Faye library to handle the boilerplate and manage connections for you.
+Here's how you'd do it in node.js using both the `faye`, and `axios` npm packages:
```
Comment (not-so-smart) [inline push.md]:
```suggestion
Here's how you'd do it in Node.js using both the `faye` and `axios` npm packages:
```

File: push.md [hunk 4]
```diff
@@ -1,27 +1,201 @@
...
+                ws.emit('message', msg);
+            });
+        }
+    }
+
+    let dms = await axios.get(`https://api.groupme.com/v3/chats?token=${GroupmeAPItoken}`);
+    dms = dms.data.response;
+    for (let i = 0; i < dms.length; i++) {
+        // subscribe to '/direct_message/:dm_id' for this DM channel.
+        // note that for some wierd reason, the REST API puts a '+' between the two
```
Comment (not-so-smart) [inline push.md]:
```suggestion
        // note that for some weird reason, the REST API puts a '+' between the two 
```

File: push.md [hunk 5]
```diff
@@ -1,27 +1,201 @@
...
```
Comment (not-so-smart) [inline push.md]:
```suggestion
        // note that for some weird reason, the REST API puts a '+' between the two 
```

**Review by not-so-smart — CHANGES_REQUESTED** [review]
This is really awesome stuff! I only have a couple of very minor suggestions, and then this should be good to go! I trust that all the scripts have been tested recently.

File: push.md (hunk 2 above)
Comment (AlureonTheVirus) [reply push.md:14]: Yep, they behave exactly the same except now they contain a `reaction` parameter to define which reaction was applied.

Comment (not-so-smart) [reply push.md:14]: Awesome, good to know. In that case I think this is all set to merge 👍

**Review by not-so-smart — APPROVED** [review]

**Pull request merged**